src/wazuh_modules/task_manager/wm_task_manager_analyze.c

```diff
@@ -93,117 +131,129 @@ cJSON* wm_task_manager_analyze_task(const cJSON *task_object, int *error_code) {
-    int result = 0;
-    char *status_result = NULL;
-
-    if (!strcmp(task_manager_commands_list[WM_TASK_UPGRADE], command) || !strcmp(task_manager_commands_list[WM_TASK_UPGRADE_CUSTOM], command)) {
+    int task_id = OS_INVALID;
 
-        if (agent_id != OS_INVALID) {
-            // Insert upgrade task into DB
-            if (task_id = wm_task_manager_insert_task(agent_id, task_manager_modules_list[WM_TASK_UPGRADE_MODULE], command), task_id == OS_INVALID) {
-                *error_code = WM_TASK_DATABASE_ERROR;
-            } else {
-                response = wm_task_manager_parse_data_response(WM_TASK_SUCCESS, agent_id, task_id, status);
-            }
+    if (agent_id != OS_INVALID) {
+        // Insert upgrade task into DB
+        if (task_id = wm_task_manager_insert_task(agent_id, module, command), task_id == OS_INVALID) {
+            *error_code = WM_TASK_DATABASE_ERROR;
         } else {
-            *error_code = WM_TASK_INVALID_AGENT_ID;
-            response = wm_task_manager_parse_data_response(WM_TASK_INVALID_AGENT_ID, agent_id, task_id, status);
+            response = wm_task_manager_parse_data_response(WM_TASK_SUCCESS, agent_id, task_id, NULL);
         }
+    } else {
+        *error_code = WM_TASK_INVALID_AGENT_ID;
+        response = wm_task_manager_parse_data_response(WM_TASK_INVALID_AGENT_ID, agent_id, task_id, NULL);
+    }
 
-    } else if (!strcmp(task_manager_commands_list[WM_TASK_UPGRADE_GET_STATUS], command)) {
+    return response;
+}
 
-        if (agent_id != OS_INVALID) {
-            // Get upgrade task status
-            if (result = wm_task_manager_get_task_status(agent_id, task_manager_modules_list[WM_TASK_UPGRADE_MODULE], &status_result), result == OS_INVALID) {
-                *error_code = WM_TASK_DATABASE_ERROR;
-            } else if (result) {
-                *error_code = result;
-                response = wm_task_manager_parse_data_response(result, agent_id, task_id, status);
-            } else {
-                response = wm_task_manager_parse_data_response(WM_TASK_SUCCESS, agent_id, task_id, status_result);
-            }
+STATIC cJSON* wm_task_manager_command_upgrade_get_status(int *error_code, int agent_id) {
+    cJSON *response = NULL;
+    int result = 0;
+    char *status_result = NULL;
+
+    if (agent_id != OS_INVALID) {
+        // Get upgrade task status
+        if (result = wm_task_manager_get_upgrade_task_status(agent_id, &status_result), result == OS_INVALID) {
+            *error_code = WM_TASK_DATABASE_ERROR;
+        } else if (result) {
+            *error_code = result;
+            response = wm_task_manager_parse_data_response(result, agent_id, OS_INVALID, NULL);
         } else {
-            *error_code = WM_TASK_INVALID_AGENT_ID;
-            response = wm_task_manager_parse_data_response(WM_TASK_INVALID_AGENT_ID, agent_id, task_id, status);
+            response = wm_task_manager_parse_data_response(WM_TASK_SUCCESS, agent_id, OS_INVALID, status_result);
```

Comment:
In this branch `error_code` is not modified at all. This works since the `error_code` is initialized to 0 in the caller function, but y suggest setting error_code to `WM_TASK_SUCCES ` in case that value changes in a future. That way, the returned error does not depend of the initial value of the variable.

Please check also this for `wm_task_manager_command_upgrade`